talisker/requests.py

```diff
@@ -189,8 +189,11 @@ def request(method, url, **kwargs):
             ctx.metric_host_name = kwargs.pop('metric_host_name', None)
             return func(method, url, **kwargs)
         finally:
-            del ctx.metric_api_name
-            del ctx.metric_host_name
+            try:
+                del ctx.metric_api_name
+                del ctx.metric_host_name
+            except Exception:
```

Comment:
I got a test failure at some point trying to delete the attribute. I'm not 100% certain, but I think it is because the underlying context object might have been been reset in the certain cases. I think I encountered it when the requests session was erroring due to shared between processes.

Either way, deleting the attribute should never cause an actual failure, hence the protection. I'll add a comment to that effect.

Reply:
Thanks. Yep, I was wondering if this is you encountering something new or a protection against app code (for example) changing the context or both.